crates/sui-graphql-rpc/src/types/transaction_block_kind/programmable.rs

```diff
@@ -0,0 +1,486 @@
+        }
+
+        Ok(connection)
+    }
+
+    /// The transaction commands, executed sequentially.
+    async fn transaction_connection(
+        &self,
+        first: Option<u64>,
+        after: Option<String>,
+        last: Option<u64>,
+        before: Option<String>,
+    ) -> Result<Connection<String, ProgrammableTransaction>> {
+        // TODO: make cursor opaque (currently just an offset).
+        validate_cursor_pagination(&first, &after, &last, &before).extend()?;
+
+        let total = self.0.commands.len();
+
+        let mut lo = if let Some(after) = after {
+            1 + after
+                .parse::<usize>()
+                .map_err(|_| Error::InvalidCursor("Failed to parse 'after' cursor.".to_string()))
+                .extend()?
+        } else {
+            0
+        };
+
+        let mut hi = if let Some(before) = before {
+            before
+                .parse::<usize>()
+                .map_err(|_| Error::InvalidCursor("Failed to parse 'before' cursor.".to_string()))
+                .extend()?
+        } else {
+            total
+        };
+
+        let mut connection = Connection::new(false, false);
+        if hi <= lo {
+            return Ok(connection);
+        }
+
+        // If there's a `first` limit, bound the upperbound to be at most `first` away from the
+        // lowerbound.
+        if let Some(first) = first {
+            let first = first as usize;
+            if hi - lo > first {
+                hi = lo + first;
+            }
+        }
+
+        // If there's a `last` limit, bound the lowerbound to be at most `last` away from the
+        // upperbound.  NB. This applies after we bounded the upperbound, using `first`.
+        if let Some(last) = last {
+            let last = last as usize;
+            if hi - lo > last {
+                lo = hi - last;
+            }
+        }
+
+        connection.has_previous_page = 0 < lo;
+        connection.has_next_page = hi < total;
+
+        for (idx, cmd) in self.0.commands.iter().enumerate().skip(lo).take(hi - lo) {
+            let input = ProgrammableTransaction::from(cmd.clone());
+            connection.edges.push(Edge::new(idx.to_string(), input));
+        }
+
+        Ok(connection)
+    }
+}
+
+#[ComplexObject]
+impl OwnedOrImmutable {
+    async fn object(&self, ctx: &Context<'_>) -> Result<Option<Object>> {
+        ctx.data_unchecked::<PgManager>()
+            .fetch_obj(self.address, Some(self.version))
+            .await
+            .extend()
+    }
+}
+
+#[ComplexObject]
+impl Receiving {
+    async fn object(&self, ctx: &Context<'_>) -> Result<Option<Object>> {
+        ctx.data_unchecked::<PgManager>()
+            .fetch_obj(self.address, Some(self.version))
+            .await
+            .extend()
+    }
+}
+
+/// A call to either an entry or a public Move function.
+#[Object]
+impl MoveCallTransaction {
+    /// The storage ID of the package the function being called is defined in.
```

Comment:
Pardon my ignorance, but why is this called `storage` ID? Is it the package is stored at this address?

Reply:
The storage ID is the ID that you can go look up on-chain to find the package that defines this function.  This is as opposed to the "runtime ID" which is the ID that the function believes itself to be at when you load its module and ask it "What is my Package ID?".  The two are different for functions in upgraded packages.